File: apps/social/test/lib/chain-goal-attachments.test.ts
import { describe, expect, it } from "vitest";
import {
  buildChainGoalWithAttachments,
  sanitizeAttachmentLabel,
  sanitizeTeamJobFileName,
} from "../../src/lib/chain-goal-attachments.js";

describe("buildChainGoalWithAttachments", () => {
  it("returns trimmed goal when there are no attachments", () => {
    expect(buildChainGoalWithAttachments("  Hello world  ", [])).toBe("Hello world");
  });

  it("puts short labels first in the Attachments block", () => {
    const goal = buildChainGoalWithAttachments("Research the brief", [
      {
        relativePath: "imports/team-jobs/tj_abc/a.pdf",
        label: "brief",
      },
      {
        relativePath: "imports/team-jobs/tj_abc/b.csv",
      },
    ]);
    expect(goal).toBe(
      [
        "Research the brief",
        "",
        "Attachments:",
        "- [brief] imports/team-jobs/tj_abc/a.pdf",
        "- imports/team-jobs/tj_abc/b.csv",
      ].join("\n"),
    );
  });

  it("skips attachments without a path", () => {
    expect(
      buildChainGoalWithAttachments("Goal only", [
        { relativePath: "  ", label: "ignored" },
      ]),
    ).toBe("Goal only");
  });
});

describe("sanitizeAttachmentLabel", () => {
  it("trims, strips brackets/newlines, and caps length", () => {
    expect(sanitizeAttachmentLabel("  source brief  ")).toBe("source brief");
    expect(sanitizeAttachmentLabel("[brief]\ndata")).toBe("brief data");
    expect(sanitizeAttachmentLabel("x".repeat(50))?.length).toBe(40);
    expect(sanitizeAttachmentLabel("   ")).toBeUndefined();
  });
});

describe("sanitizeTeamJobFileName", () => {
  it("strips path separators", () => {
    expect(sanitizeTeamJobFileName("../../evil/name.pdf")).toBe(".._.._evil_name.pdf");
    expect(sanitizeTeamJobFileName("plain.txt")).toBe("plain.txt");
  });
});

File: apps/envoygo/test/chain_goal_attachments_test.dart
import 'package:envoygo/chain_goal_attachments.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  group('buildChainGoalWithAttachments', () {
    test('returns trimmed goal when there are no attachments', () {
      expect(
        buildChainGoalWithAttachments('  Hello world  ', const []),
        'Hello world',
      );
    });

    test('puts short labels first in the Attachments block', () {
      final goal = buildChainGoalWithAttachments('Research the brief', [
        const ChainGoalAttachment(
          relativePath: 'imports/team-jobs/tj_abc/a.pdf',
          label: 'brief',
        ),
        const ChainGoalAttachment(
          relativePath: 'imports/team-jobs/tj_abc/b.csv',
        ),
      ]);
      expect(
        goal,
        [
          'Research the brief',
          '',
          'Attachments:',
          '- [brief] imports/team-jobs/tj_abc/a.pdf',
          '- imports/team-jobs/tj_abc/b.csv',
        ].join('\n'),
      );
    });

    test('skips attachments without a path', () {
      expect(
        buildChainGoalWithAttachments('Goal only', [
          const ChainGoalAttachment(relativePath: '  ', label: 'ignored'),
        ]),
        'Goal only',
      );
    });
  });

  group('sanitizeAttachmentLabel', () {
    test('trims, strips brackets/newlines, and caps length', () {
      expect(sanitizeAttachmentLabel('  source brief  '), 'source brief');
      expect(sanitizeAttachmentLabel('[brief]\ndata'), 'brief data');
      expect(sanitizeAttachmentLabel('x' * 50)?.length, 40);
      expect(sanitizeAttachmentLabel('   '), isNull);
    });
  });

  group('sanitizeTeamJobFileName', () {
    test('strips path separators', () {
      expect(
        sanitizeTeamJobFileName('../../evil/name.pdf'),
        '.._.._evil_name.pdf',
      );
      expect(sanitizeTeamJobFileName('plain.txt'), 'plain.txt');
    });
  });
}
